sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/GroupIntoBatchesTest.java

```diff
@@ -125,6 +126,45 @@ public Void apply(Iterable<KV<String, Iterable<String>>> input) {
     pipeline.run();
   }
 
+  @Test
+  @Category({NeedsRunner.class, UsesTimersInParDo.class, UsesStatefulParDo.class})
+  public void testWithShardedKeyInGlobalWindow() {
+    // Since with default sharding, the number of subshards of of a key is nondeterministic, create
```

Comment:
Could we go with a larger batch size (say 5 or 10) and also verify that most batches are of the expected size?

Reply:
Done.